src/main/src/main/java/org/geoserver/security/decorators/SecuredFeatureTypeInfo.java

```diff
@@ -103,14 +107,41 @@ public FeatureType getFeatureType() throws IOException {
             ProgressListener listener, Hints hints) throws IOException {
         final FeatureSource<? extends FeatureType, ? extends Feature> fs =
                 delegate.getFeatureSource(listener, hints);
-
-        if (policy.level == AccessLevel.METADATA) {
+        Request request = Dispatcher.REQUEST.get();
+        if (policy.level == AccessLevel.METADATA && !isGetCapabilities(request)) {
             throw SecureCatalogImpl.unauthorizedAccess(this.getName());
         } else {
-            return SecuredObjects.secure(fs, policy);
+            return SecuredObjects.secure(fs, computeWrapperPolicy(request));
         }
     }
 
+    /**
+     * Checks if current request is GetCapabilities and returns a new WrapperPolicy with attributes
+     * read allowed to compute the dimensions values. If current request is not a GetCapabilities
+     * one, returns the same policy without changes.
+     */
+    private WrapperPolicy computeWrapperPolicy(Request request) {
+        if (isGetCapabilities(request) && policy.getLimits() instanceof VectorAccessLimits) {
```

Comment:
I would recommend to add an INFO LOG here explaining what's happening.

Reply:
Makes sense, adding it.